deeplearning4j-keras/src/main/java/org/deeplearning4j/keras/NeuralNetworkBatchLearner.java

```diff
@@ -0,0 +1,41 @@
+package org.deeplearning4j.keras;
+
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.indexing.INDArrayIndex;
+import org.nd4j.linalg.indexing.NDArrayIndex;
+
+@Slf4j
+public class NeuralNetworkBatchLearner {
+
+    private INDArrayIndex[] createSlicingIndexes(int length) {
+        INDArrayIndex[] ndIndexes = new INDArrayIndex[length];
+        for (int i = 0; i < ndIndexes.length; i++) {
+            ndIndexes[i] = NDArrayIndex.all();
+        }
+        return ndIndexes;
+    }
+
+    public void fitInBatches(MultiLayerNetwork multiLayerNetwork, INDArray features, INDArray labels, int batchSize) {
+        final INDArrayIndex[] ndIndexes = createSlicingIndexes(features.shape().length);
+
+        int begin = 0;
+        int samplesCount = features.size(0);
+
```

Comment:
This feels wrong to me. Does it have to be an in memory object? We should be consistent and try to to use iterators if possible.